services/ui-src/src/components/fields/DropdownField.tsx

```diff
@@ -24,6 +25,26 @@ export const DropdownField = ({
   // get form context and register field
   const form = useFormContext();
   form.register(name);
+  const { report } = useContext(ReportContext);
+
+  let dropdownOptions = [];
+  if (typeof options === "string") {
+    const dynamicOptionValues = report?.fieldData[options];
+    if (dynamicOptionValues) {
+      const fieldOptions = dynamicOptionValues.map((value: string) => ({
+        label: value,
+        value: value,
+      }));
```

Comment:
When we change this to loop through objects instead I think it should look like this, using the id as the value because it is immutable, and using the mutable name as the label.
```suggestion
      const fieldOptions = dynamicOptionValues.map((option: EntityShape) => ({
        label: option.name,
        value: option.id,
      }));
```